### PR TITLE
Makes everything smooth as a pomf's behind.

### PR DESCRIPTION
**Pull request by EdXbtkOEwGw**
This is something that needs to be experienced first-hand to fully understand it.
I also couldn't add a new argument to step() since it's a BYOND proc, so I'm just setting the glide_size before every step().

https://a.uguu.se/Y7JuBWYR6zst_2018-01-08_21-47-40.mp4
https://user-images.githubusercontent.com/33260604/34704533-711ae136-f4d8-11e7-8eb7-0a723ddfe9aa.gif

:cl:
 * rscadd: Properly implemented glide_size on a lot of things. This means, for example, dragging a corpse won't make it look like it's teleporting everywhere, ghosts won't rollerskate in the air after haunting a slow person, plus smooth conveyor belts, smooth space inertia, and other things. If you still notice old no-glidesize behavior of stuff teleporting around, make a bug report.

### CODE REVIEW

**Comment by chatsonf** [comment]
You are a god.
I finally can stop teleporting while being dragged and walking

**Review by gbasood — APPROVED** [review]

**Comment by ghost** [comment]
Nice.

**Comment by gbasood** [comment]
actually there might be a better way than setting glide size before every step but I don't know enough about movecode anymore

**Comment by EdXbtkOEwGw** [comment]
According to BYOND, the third argument of step() should do exactly this but from what I could see in testing it didn't do nothin'

**Comment by ghost** [comment]
where have you been when vg became most advanced codebase

**Comment by MadmanMartian** [comment]
mother of god

**Comment by Zennyization** [comment]
Probably the best PR I've seen in a long time.

**Comment by SkowronX** [comment]
yeus

**Comment by Exxion** [comment]
For the record, the third arg to `step()` is for pixel movement

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -444,60 +448,18 @@
 				else
 					mob.forceEnter(get_step(mob, direct))
 					mob.dir = direct
-			if(isobserver(mob))
-				var/mob/dead/observer/observer = mob
-				mob.delayNextMove(observer.movespeed)
-			else
-				mob.delayNextMove(1)
-		if(INCORPOREAL_NINJA)
```
Comment (MadmanMartian) [inline code/modules/mob/mob_movement.dm:452]: Making a note of lost antag code

Flimbeams

**Pull request merged**